src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -0,0 +1,389 @@
+
+        [Required]
+        public string TargetFrameworkMoniker { get; set; }
+
+        public string RuntimeIdentifier { get; set; }
+
+        public bool MarkPackageReferencesAsExternallyResolved { get; set; }
+
+        [Output]
+        public ITaskItem[] Analyzers { get; private set; }
+
+        [Output]
+        public ITaskItem[] ContentFiles { get; private set; }
+
+        [Output]
+        public ITaskItem[] CompileTimeAssemblies { get; private set; }
+
+        [Output]
+        public ITaskItem[] FrameworkAssemblies { get; private set; }
+
+        [Output]
+        public ITaskItem[] NativeLibraries { get; private set; }
+
+        [Output]
+        public ITaskItem[] ResourceAssemblies { get; private set; }
+
+        [Output]
+        public ITaskItem[] RuntimeAssemblies { get; private set; }
+
+        [Output]
+        public ITaskItem[] RuntimeTargets { get; private set; }
+
+        [Output]
+        public ITaskItem[] TransitiveProjectReferences { get; private set; }
+
+        private NuGetPackageResolver _packageResolver;
+
+        protected override void ExecuteCore()
+        {
+            var lockFile = new LockFileCache(BuildEngine4).GetLockFile(ProjectAssetsFile);
+            _packageResolver = NuGetPackageResolver.CreateResolver(lockFile, ProjectPath);
+
+            var targetFramework = NuGetUtils.ParseFrameworkName(TargetFrameworkMoniker);
+            var compileTimeTarget = lockFile.GetTargetAndThrowIfNotFound(targetFramework, runtime: null);
+            var runtimeTarget = lockFile.GetTargetAndThrowIfNotFound(targetFramework, RuntimeIdentifier);
+
+            Analyzers = RaiseAnalyzers(
+                lockFile,
+                runtimeTarget);
+
+            CompileTimeAssemblies = RaisePackageAssets(
+                compileTimeTarget,
+                package => package.CompileTimeAssemblies,
+                setup: (asset, item) =>
+                {
+                    item.SetMetadata("Private", "false");
+                    item.SetMetadata("HintPath", item.ItemSpec);
+
+                    if (MarkPackageReferencesAsExternallyResolved)
+                    {
+                        item.SetMetadata("ExternallyResolved", "true");
+                    }
+                });
+
+            if (!DisableFrameworkAssemblies)
+            {
+                FrameworkAssemblies = RaiseFrameworkAssemblies(
+                    compileTimeTarget,
+                    setup: (asset, item) =>
+                     {
+                         item.SetMetadata(MetadataKeys.Pack, "false");
+                         item.SetMetadata(MetadataKeys.Private, "false");
+                         item.SetMetadata(MetadataKeys.NuGetIsFrameworkReference, "true");
+                     });
+            }
+
+            ContentFiles = RaisePackageAssets(
+                runtimeTarget,
+                p => p.ContentFiles,
+                filter: asset => !string.IsNullOrEmpty(asset.PPOutputPath),
```

Comment:
Oops. I forgot to loop back to this. The thing is that we only process the PPOutputPath files in the common case because nuget writes content items directly to generated props otherwise. There's a bool to override it though. I decided to revert the optimization, but now I'm thinking about adding it back. I need to add a content file test still.

Reply:
Fixed.

Reply:
Fixed.